crates/forge-runner/src/build_trace_data.rs

```diff
@@ -55,20 +60,57 @@ pub fn build_profiler_call_trace(
         nested_calls: value
             .nested_calls
             .iter()
-            .map(|c| build_profiler_call_trace_node(c, contracts_data))
+            .map(|c| build_profiler_call_trace_node(c, contracts_data, test_artifacts_path))
             .collect(),
-        vm_trace,
+        cairo_execution_info,
+    }
+}
+
+fn build_cairo_execution_info(
+    entry_point: &ProfilerCallEntryPoint,
+    vm_trace: Option<Vec<ProfilerTraceEntry>>,
+    contracts_data: &ContractsData,
+    test_artifacts_path: &Utf8Path,
+) -> Option<CairoExecutionInfo> {
+    let cairo_execution_info_available = vm_trace.is_some() && entry_point.contract_name.is_some();
+
+    if cairo_execution_info_available {
```

Comment:
This should not be decision of forge, this is profiler abstraction layer. Forge should just provide all available information and its profiler role to determine if it can interpret it or not.